core/controllers/resources.py

```diff
@@ -40,27 +40,27 @@ def get(self, generator_id):
             raise self.PageNotFoundException
 
 
-class ImageHandler(base.BaseHandler):
+class ImageDevHandler(base.BaseHandler):
     """Handles image retrievals (only in dev -- in production, image files are
     served from GCS).
     """
 
     _IMAGE_PATH_PREFIX = 'image'
 
     @acl_decorators.open_access
-    def get(self, exploration_id, encoded_filepath):
+    def get(self, exploration_id, filename):
         """Returns an image.
 
         Args:
             exploration_id: the id of the exploration.
-            encoded_filepath: a string representing the image filepath. This
+            filename: a string representing the image filename. This
               string is encoded in the frontend using encodeURIComponent().
         """
         if not constants.DEV_MODE:
             raise self.PageNotFoundException
         try:
-            filepath = urllib.unquote(encoded_filepath)
-            file_format = filepath[(filepath.rfind('.') + 1):]
+            filename = urllib.unquote(filename)
```

Comment:
Try not to redefine a variable name. I think it's better to keep the input arg as encoded_filename; it's then always clear what each variable means.

Ditto below.

Reply:
Good point, thanks!